client/src/firebase/functions/src/index.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-throw-literal */
+/* eslint-disable linebreak-style */
+/* eslint-disable max-len */
+/**
+ * Import function triggers from their respective submodules:
+ *
+ * import {onCall} from "firebase-functions/v2/https";
+ * import {onDocumentWritten} from "firebase-functions/v2/firestore";
+ *
+ * See a full list of supported triggers at https://firebase.google.com/docs/functions
+ */
+
+import * as functions from "firebase-functions";
+import * as logger from "firebase-functions/logger";
+import * as admin from "firebase-admin";
+import * as serviceAccount from "./callforhelp-37002-firebase-adminsdk-7ivya-e3e03f01cb.json";
+
+const DATABASE_URL = "https://callforhelp-37002-default-rtdb.firebaseio.com";
+
+admin.initializeApp({
+  credential: admin.credential.cert(serviceAccount as admin.ServiceAccount),
+  databaseURL: DATABASE_URL,
+});
+
+// Start writing functions
+// https://firebase.google.com/docs/functions/typescript
+
+export const signUp = functions.https.onCall(async (request) => {
+  logger.log(request);
+  const {email, password, name}: { email?: string; password?: string; name?: string } =
+    request;
+
+  if (!email || !password || !name) {
+    logger.error("The function requires an email, a password and a name: " + request);
+    throw new functions.https.HttpsError(
+      "invalid-argument",
+      "צריך אימייל, סיסמא ואימייל"
+    );
+  }
+
+  if (request.auth) {
+    logger.error("The user must be unauthenticated in order to sign up: " + request);
+    throw new functions.https.HttpsError(
+      "failed-precondition",
+      "אי אפשר להירשם בתור משתמש מחובר"
+    );
+  }
+
+  const passwordRegex = new RegExp(
+    /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&_])[A-Za-z\d@$!%*?&_]{8,20}$/
+  );
+  const emailRegex = new RegExp(/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/);
+
+  try {
+    if (!passwordRegex.test(password)) {
+      logger.error("The password should include at least 8 characters, 1 uppercase letter, 1 lowercase letter, 1 number and one symbol from the following: @$!%*?&_: " + request);
+      throw new functions.https.HttpsError(
+        "invalid-argument",
+        "הסיסמא צריכה להכיל לפחות אות קטנה אחת, אות גדולה אחת, מספר וסימן מיוחד (@,$,!,%,*,?,&,_)"
+      );
+    }
+  } catch (error) {
+    logger.error("Error checking regex: " + error + ", " + request);
+    throw new functions.https.HttpsError("invalid-argument", "הסיסמא צריכה להכיל לפחות אות קטנה אחת, אות גדולה אחת, מספר וסימן מיוחד (@,$,!,%,*,?,&,_)");
+  }
+  if (password.length < 8 || password.length > 20) {
```

Comment:
isn't that included in the regex check?

Reply:
נכון, הסדר צריך להיות הפוך בשביל שההודעה תהיה יותר ברורה